src/PrestaShopBundle/Form/Admin/Type/EventListener/AddDisablingSwitchListener.php

```diff
@@ -0,0 +1,157 @@
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Type\EventListener;
+
+use PrestaShopBundle\Form\Admin\Extension\DisablingSwitchExtension;
+use PrestaShopBundle\Form\Admin\Type\DisablingSwitchType;
+use PrestaShopBundle\Form\FormCloner;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\Event\PreSetDataEvent;
+use Symfony\Component\Form\Exception\InvalidConfigurationException;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\Form\FormInterface;
+use TypeError;
+
+/**
+ * This listener is used by the DisablingSwitchExtension to automatically add the DisablingSwitchType to the parent,
+ * it is not possible to access the parent builder in the extension which is why this operation is delayed on the
+ * PRE_SET_DATA event.
+ *
+ * The switch state and the associated input's disabled state are automatically computed based on the input's data,
+ * if it matches the disabled_value then the field is considered as disabled and the switch is turned off.
+ */
+class AddDisablingSwitchListener implements EventSubscriberInterface
+{
+    public const TOGGLE_DATA_ATTRIBUTE = 'data-toggled-by';
+
+    private const DISABLED_VALUE = '0';
+    private const ENABLED_VALUE = '1';
+
+    /**
+     * @var FormCloner
+     */
+    private $formCloner;
+
+    public function __construct(FormCloner $formCloner)
+    {
+        $this->formCloner = $formCloner;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            FormEvents::PRE_SET_DATA => 'addDisablingSwitch',
+        ];
+    }
+
+    /**
+     * @param PreSetDataEvent $event
+     */
+    public function addDisablingSwitch(PreSetDataEvent $event): void
+    {
+        $form = $event->getForm();
+        $parent = $form->getParent();
+        if (null === $parent) {
+            throw new InvalidConfigurationException(sprintf('You cannot set the option %s on a root form.', DisablingSwitchExtension::SWITCH_OPTION));
+        }
+
+        $disablingFieldName = DisablingSwitchExtension::FIELD_PREFIX . $form->getName();
+        if ($parent->has($disablingFieldName)) {
+            return;
+        }
+
+        $disabledValue = $form->getConfig()->getOption(DisablingSwitchExtension::DISABLED_VALUE_OPTION);
+        if (is_callable($disabledValue)) {
+            try {
+                $shouldBeDisabled = $disabledValue($event->getData(), $event->getForm());
+            } catch (TypeError $typeError) {
+                throw new InvalidConfigurationException(
+                    'The callable provided for disabled_value option seems invalid, its prototype should be compatible with function($data, FormInterface $form): void And $data is usually nullable',
```

Comment:
Long line, no issue with CI?

Reply:
Actually I don't think we have a Cs fixer rule for max columns, only recommendations